dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/WebLogEventConsumer.kt

```diff
@@ -6,13 +6,51 @@
 
 package com.datadog.android.rum.webview
 
+import com.datadog.android.core.internal.persistence.DataWriter
 import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.log.LogAttributes
+import com.datadog.android.log.model.LogEvent
 import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
 
-internal class WebLogEventConsumer {
+internal class WebLogEventConsumer(
+    private val userLogsWriter: DataWriter<LogEvent>,
+    private val internalLogsWriter: DataWriter<LogEvent>,
+    private val rumContextProvider: WebRumEventContextProvider
+) {
 
-    fun consume(event: JsonObject) {
-        // just to make the code compile without warning. This will be removed in the next PR.
-        sdkLogger.i(event.toString())
+    fun consume(event: JsonObject, eventType: String) {
+        map(event)?.let {
+            if (eventType == INTERNAL_LOG_EVENT_TYPE) {
+                internalLogsWriter.write(it)
+            } else {
+                userLogsWriter.write(it)
+            }
+        }
+    }
+
+    private fun map(event: JsonObject): LogEvent? {
+        return try {
+            val logEvent = LogEvent.fromJson(event.toString())
+            val rumContext = rumContextProvider.getRumContext()
+            if (rumContext != null) {
```

Comment:
We can either do this or we will have to change our Log `DataWriter` to accept pure JsonObjects. I would rather do this than that and see how it goes. In case the deserialization fails for the reason of some missing attributes we could change all the attributes in the log event internal json-schema as optional.